### PR TITLE
ci: Override path for BSPs in examples

### PR DESCRIPTION
**Pull request by tore-espressif**
By using local components in examples, we make sure that we compile actual code from the specific PR (rather than already uploaded component in the component service)

### CODE REVIEW

**Comment by tore-espressif** [comment]
cc @kumekay

File: examples/display/main/idf_component.yml [hunk 1]
```diff
@@ -2,6 +2,6 @@ description: BSP Display example
 
 dependencies:
   idf: ">=4.4"
-  esp_wrover_kit: ">=1.0.0"
-  # esp32_s2_kaluga_kit: ">=1.0.0"
-  # esp-box: ">=1.0.0"
+  esp_wrover_kit:
+    version: ">=1.0.0"
```
Comment (kumekay) [inline examples/display/main/idf_component.yml:6]: Want be better to limit versions to compatible ones?
```suggestion
    version: "^1.0.0"
```

**Review by kumekay — COMMENTED** [review]
LGTM, but I suggest uploading to staging first, to be sure it works as expected

**Comment by espzav** [comment]
I know, that we was solving it, but I don't like see this in example YML files. Because what customer will do? He will copy the example and it wouldn't working. Then he will copy all components from BSP instead of use it from component manager. 

I know, that there are no other way. Maybe some note in YML files.

**Comment by espzav** [comment]
@kumekay thank you for clarification. Then it is ok for me 👍

**Review by espzav — APPROVED** [review]

**Comment by kumekay** [comment]
@tore-espressif I'm not too fond of the idea of fallback, to me, it's too magical, nobody reads warnings. I suggest an error with clear suggestions on what to remove from which manifest to continue. Keeping the `override_path` in the manifest forever can be error prone, because project may depend on relative path.

**Comment by espzav** [comment]
I agree with @kumekay, there should be error.

**Comment by tore-espressif** [comment]
> there should be error

fair enough. Can we merge it as it is now?

**Comment by kumekay** [comment]
@tore-espressif LGTM

**Pull request merged**